adapters/admatic/admatic.go

```diff
@@ -32,85 +31,67 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 }
 
 func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errs []error
 
-	host := ""
-
+	requestCopy := *request
 	for _, imp := range request.Imp {
```

Comment:
Given that this loop now creates a separate request for each imp, please either update an exemplary test case to test the multiple imp case or create a supplemental test case to cover it.

Additionally, if it is easy, I suggest adding a supplemental multi-imp test case where one imp has an error and the other does not. I think you can easily force an error when building the endpoint for a given imp by specifying an invalid type like an empty object for `host`.

Reply:
Hello @bsardo,
It returns an error when there is an invalid host or networkId in an imp array.

Example: 
Invalid request: request.imp[1].ext.prebid.bidder.admatic failed validation.
host: Invalid type. Expected: string, given: integer

Reply:
Hi @bakicam, PBS core will perform bidder param validation upstream. However, I'm suggesting that you first add an exemplary test case for when there are multiple well-formed imps, and then a second supplemental one that demonstrates how your adapter works when there are multiple imps where one of the imps is malformed. You don't have any multiple imp tests at the moment.
If your adapter is given a request with multiple imps and one of the imps has an error, your adapter appears to discard the malformed imp while making a bid request for the well-formed imp.

Given that adapter JSON tests are scoped to the adapter code, bidder param validation should be bypassed. For the multiple imp with one malformed case, the following test can be added:
[multiple-imps-with-error.json](https://github.com/user-attachments/files/16069543/multiple-imps-with-error.json)

Reply:
@bakicam the new file looks good. As I mentioned, you might also want to add `exemplary/multiple-imps.json` that does not involve any errors and demonstrates that your adapter sends two separate requests, one for each imp, to your adapter. Last suggestion from me 😄

Reply:
Hello @bsardo , thank you very much for your valuable suggestions and help. I also added this file. 😊